### PR TITLE
adr authorization

### PR DESCRIPTION
**Pull request by TopSwagCode**

### CODE REVIEW

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Metric naming convention
+
+* Status: Accepted
+* Deciders: @tnickelsen, @ckr123, @sahma19, @TopSwagCode, @martinhenningjensen
```
Comment (sahma19) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:4]: Deciders were @sahma19 @TopSwagCode @martinhenningjensen @rvplauborg @Sondergaard

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+* Date: 2025-01-01
```
Comment (sahma19) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:6]: Date was 2024-07-02

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Auth0 / other cloud / hosted 3rd party solutions. Unknown / long Vetting proccess. Would block the project for long and may never be vetted for use at Energinet.
+
+Hydra / Keyclock / other self hosted solutions. Handling the security aspects of running an identity provider. Having java / golang knownledge to help with customization of these products. Handling backup of data.
+
+Azure B2C. Other teams has used this before. Needing to learning custom policies xml to make customizations. Already vetted products that we can start using today.
+
+## Decision Outcome
+
+We have choosen to go with Azure B2C, because it's a vetted secure hosted solution, that is already used within the company and most likely going to be the standard in the future. We had acritect in on this decision and they have scored all the solutions with the team. Azure B2C was the clear winner.
```
Comment (sahma19) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:30]: Does all this correspond with what is written here? Should it?

https://energinet.atlassian.net/wiki/spaces/EnergyOrigin/pages/640974851/OAuth2+PoC

**Review by sahma19 — CHANGES_REQUESTED** [review]

File: doc/architecture/adr/0028-Wallet-Proxy.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Metric naming convention
```
Comment (ckr123) [inline doc/architecture/adr/0028-Wallet-Proxy.md:1]: Wallet Proxy*

File: doc/architecture/adr/0028-Wallet-Proxy.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+* Status: Accepted
+* Deciders: @tnickelsen, @ckr123, @sahma19, @TopSwagCode, @martinhenningjensen
+
+* Date: 2025-01-01
```
Comment (ckr123) [inline doc/architecture/adr/0028-Wallet-Proxy.md:6]: Wrong date

File: doc/architecture/adr/0027-Authentication-and-Authorization.md (hunk 3 above)
Comment (TopSwagCode) [reply doc/architecture/adr/0027-Authentication-and-Authorization.md:30]: You tell me? :D I wrote from my memory.

**Review by martinhenningjensen — APPROVED** [review]

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+* Date: 2024-29-08
+
+---
+
+## Context and Problem Statement
+
+We need a way to handle authentication and authorization of these use cases:
+
+* A mitid user needs to be able to securely view and do actions on their data.
+* A mitid needs to be able to give 3rd party system access to work on their data.
```
Comment (martinhenningjensen) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:15]: mitid _user_ needs

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+* A user needs to be able to give 3rd party mitid user access to work on their data.
```
Comment (martinhenningjensen) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:16]: A _mitid_ user needs

File: doc/architecture/adr/0027-Authentication-and-Authorization.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+| Self service                     | 3                 | 1                        | 3                            | 3                                  |
+| Security                         | 2                 | 1                        | 2                            | 3                                  |
+| Longevity                        | 2                 | 1                        | 2                            | 3                                  |
+| Notes                            |                   |                          |                              | In preview unfortunately!          |
+| Total                            | 22                | 18                       | 23                           | 29                                 |
+
+## Decision Outcome
+
+We have choosen to go with Azure B2C, because it's a vetted secure hosted solution, that is already used within the company and most likely going to be the standard in the future. We had architect in on this decision and they have scored all the solutions with the team. Azure B2C was the clear winner.
```
Comment (martinhenningjensen) [inline doc/architecture/adr/0027-Authentication-and-Authorization.md:49]: _We included architects in the decision_

File: doc/architecture/adr/0028-Wallet-Proxy.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+We could fork wallet and maintain our own version of wallet with our authentication and authorization logic.
+
+Pro:
+* Fully customizable of logic in wallet.
+
+Con:
+* Merge hells, when wallet changes in main repository.
+* Might get out of sync with API endpoints and wallets wouldn't work with other wallet systems.
+
+Make auhtorization proxy for wallet:
```
Comment (martinhenningjensen) [inline doc/architecture/adr/0028-Wallet-Proxy.md:28]: typo: _authorization_

File: doc/architecture/adr/0028-Wallet-Proxy.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+Make a new subsystem to protect wallet behind as a proxy having custmo logic.
```
Comment (martinhenningjensen) [inline doc/architecture/adr/0028-Wallet-Proxy.md:30]: typo: _custom_

**Review by martinhenningjensen — DISMISSED** [review]

File: doc/architecture/adr/0027-Authentication-and-Authorization.md (hunk 6 above)
Comment (sahma19) [reply doc/architecture/adr/0027-Authentication-and-Authorization.md:15]: Ditto

File: doc/architecture/adr/0027-Authentication-and-Authorization.md (hunk 7 above)
Comment (sahma19) [reply doc/architecture/adr/0027-Authentication-and-Authorization.md:16]: Ditto

**Review by sahma19 — DISMISSED** [review]

File: doc/architecture/adr/0028-Wallet-Proxy.md (hunk 9 above)
Comment (TopSwagCode) [reply doc/architecture/adr/0028-Wallet-Proxy.md:28]: ?

**Review by martinhenningjensen — APPROVED** [review]

**Pull request merged**